examples/cifar10/pt/learners/cifar10_scaffold_learner.py

```diff
@@ -0,0 +1,180 @@
+
+from nvflare.apis.dxo import DXO, DataKind, MetaKey, from_shareable
+from nvflare.apis.fl_constant import ReturnCode
+from nvflare.apis.fl_context import FLContext
+from nvflare.apis.shareable import Shareable, make_reply
+from nvflare.apis.signal import Signal
+from nvflare.app_common.app_constant import AlgorithmConstants, AppConstants
+from nvflare.app_common.pt.pt_scaffold import PTScaffold
+
+
+class CIFAR10ScaffoldLearner(CIFAR10Learner, PTScaffold):
+    def __init__(
+        self,
+        dataset_root: str = "./dataset",
+        aggregation_epochs: int = 1,
+        train_task_name: str = AppConstants.TASK_TRAIN,
+        submit_model_task_name: str = AppConstants.TASK_SUBMIT_MODEL,
+        lr: float = 1e-2,
+        fedproxloss_mu: float = 0.0,
+        central: bool = False,
+        analytic_sender_id: str = "analytic_sender",
+    ):
+        """Simple Scaffold CIFAR-10 Trainer.
+        Implements the training algorithm proposed in
+        Karimireddy et al. "SCAFFOLD: Stochastic Controlled Averaging for Federated Learning"
+        (https://arxiv.org/abs/1910.06378) using functions implemented in `ScaffoldLearner`.
+
+        Args:
+            dataset_root: directory with CIFAR-10 data.
+            aggregation_epochs: the number of training epochs for a round. Defaults to 1.
+            train_task_name: name of the task to train the model.
+            submit_model_task_name: name of the task to submit the best local model.
+            lr: local learning rate. Float number. Defaults to 1e-2.
+            fedproxloss_mu: weight for FedProx loss. Float number. Defaults to 0.0 (no FedProx).
+            central: Bool. Whether to simulate central training. Default False.
+            analytic_sender_id: id of `AnalyticsSender` if configured as a client component. If configured, TensorBoard events will be fired. Defaults to "analytic_sender".
+
+        Returns:
+            a Shareable with the updated local model after running `execute()`
+            or the best local model depending on the specified task.
+        """
+
+        CIFAR10Learner.__init__(
+            self,
+            dataset_root=dataset_root,
+            aggregation_epochs=aggregation_epochs,
+            train_task_name=train_task_name,
+            submit_model_task_name=submit_model_task_name,
+            lr=lr,
+            fedproxloss_mu=fedproxloss_mu,
+            central=central,
+            analytic_sender_id=analytic_sender_id,
+        )
+        PTScaffold.__init__(self)
+
+    def initialize(self, parts: dict, fl_ctx: FLContext):
+        # Initialize super class and SCAFFOLD
+        CIFAR10Learner.initialize(self, parts=parts, fl_ctx=fl_ctx)
+        self.scaffold_init()
+
+    def local_train(self, fl_ctx, train_loader, model_global, abort_signal: Signal, val_freq: int = 0):
+        # local_train with SCAFFOLD steps
+        c_global_para, c_local_para = self.scaffold_get_params()
```

Comment:
In the method definition, it returns  `c_global_para = self.c_global.state_dict()`

If `self.c_global` does not change, we don't need to return this and pass it into `scaffold_model_update` again?

Same thing for `c_local_para`